core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/ndjsonld/NDJSONLDWriter.java

```diff
@@ -0,0 +1,113 @@
+package org.eclipse.rdf4j.rio.ndjsonld;
+
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.rio.*;
+import org.eclipse.rdf4j.rio.helpers.BasicWriterSettings;
+import org.eclipse.rdf4j.rio.helpers.BufferedGroupingRDFHandler;
+import org.eclipse.rdf4j.rio.helpers.JSONLDSettings;
+import org.eclipse.rdf4j.rio.jsonld.JSONLDWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class NDJSONLDWriter extends BufferedGroupingRDFHandler implements RDFWriter {
+
+    private final String baseURI;
+    private final Writer writer;
+    private WriterConfig writerConfig = new WriterConfig();
+
+
+    /**
+     * Creates a new NDJSONLDWriter that will write to the supplied OutputStream.
+     *
+     * @param outputStream The OutputStream to write the NDJSONLD document to.
+     */
+    public NDJSONLDWriter(OutputStream outputStream) {
+        this(outputStream, null);
+    }
+
+    public NDJSONLDWriter(OutputStream out, String baseURI) {
+        this.writer = new BufferedWriter(new OutputStreamWriter(out, StandardCharsets.UTF_8));;
+        this.baseURI = baseURI;
+    }
+
+    public NDJSONLDWriter(Writer writer) {
+        this(writer, null);
+    }
+
+    public NDJSONLDWriter(Writer writer, String baseURI) {
+        this.baseURI = baseURI;
+        this.writer = writer;
+    }
+
+    private JSONLDWriter getJSONLDWriter() {
+        JSONLDWriter jsonldWriter = new JSONLDWriter(writer, baseURI);
+        WriterConfig config = getWriterConfig();
+        config.set(BasicWriterSettings.PRETTY_PRINT, false);
+        jsonldWriter.setWriterConfig(config);
+        return jsonldWriter;
+    }
+
+    protected void processBuffer() throws RDFHandlerException {
+        // primary grouping per context.
+        for (Resource subject : getBufferedStatements().subjects()) {
+            // retrieves other statements with the same
+            // subject, and output them grouped using a JSON LD Writer
+            Iterable<Statement> statements = getBufferedStatements().getStatements(subject, null, null);
+            JSONLDWriter jsonldWriter = getJSONLDWriter();
+            jsonldWriter.startRDF();
+            for (Statement st : statements) {
+                jsonldWriter.handleStatement(st);
+            }
+            jsonldWriter.endRDF();
+            try {
+                writer.write("\n");
```

Comment:
I think System.lineSeparator() is better than "\n" here as it will do "\r\n" on Windows.